### PR TITLE
Add default_max_pods_per_node and max_pods_per_node after beta_cluster templating 

### PR DESCRIPTION
**Pull request by richardmcsong**
Relates to #174 , #170. I've transplanted the changes made by @algermissen. However, it seems like the `make generate_docs` is failing with:
```
Working in ./modules/private-cluster ...
test/make.sh: line 140: terraform_docs.sh: command not found
Warning! Exit code: 127
```

Am I misconfiguring something? I have terraform-docs 0.6.0 installed.

### CODE REVIEW

**Comment by Dev25** [comment]
@richardmcsong you need to install https://github.com/antonbabenko/pre-commit-terraform or at least have that terraform_docs script in your $PATH. 

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/209

File: autogen/cluster_regional.tf [hunk 1]
```diff
@@ -59,6 +59,7 @@ resource "google_container_cluster" "primary" {
 {% if beta_cluster %}
   enable_binary_authorization = var.enable_binary_authorization
   enable_intranode_visibility = var.enable_intranode_visibility
+  default_max_pods_per_node = var.default_max_pods_per_node
```
Comment (morgante) [inline autogen/cluster_regional.tf]: Formatting needs to be flush with line above.

File: autogen/cluster_zonal.tf [hunk 2]
```diff
@@ -204,7 +205,9 @@ resource "google_container_node_pool" "zonal_pools" {
     "initial_node_count",
     lookup(var.node_pools[count.index], "min_count", 1),
   )
-
+  {% if beta_cluster %}
+  max_pods_per_node  = lookup(var.node_pools[count.index], "max_pods_per_node", 110)
```
Comment (morgante) [inline autogen/cluster_zonal.tf]: Should this reference var. default_max_pods_per_node instead of harcoding 110?

Comment (richardmcsong) [reply autogen/cluster_zonal.tf]: great point.

var.default_max_pods_per_node is a terraform attribute for google_container_cluster though. maybe we should be looking to only set max_pods_per_node if the node pool has max_pods_per_node set in the var.node_pools map?

Comment (morgante) [reply autogen/cluster_zonal.tf]: Agreed. We can likely use the Terraform 0.12 `nil` value for this use case.

Comment (richardmcsong) [reply autogen/cluster_zonal.tf]: Great shout out, I didn't even know about that functionality! Added null as the default value for the lookup

**Pull request merged**